### PR TITLE
OpenTelemetry OTLP setup for tracing

### PR DESCRIPTION
**Pull request by mzabaluev**
Resolves: #555 #556

# Summary
- **Categories**: `networks`, `util`.

Integrate an OTLP exporter for tracing events matching the "movement_tracing" target.
The OTLP gRPC endpoint is configured with the `MOVEMENT_OTLP` environment variable.

This replaces the earlier "movement_timing" tracing layer, as analysis of OpenTelemetry should be more versatile.

# Changelog

* Added OpenTelemetry OTLP exporter, with the endpoint URL configured with the `MOVEMENT_OTLP` environment variable.
* Removed support for timing JSON logs configured with `MOVEMENT_TIMING`.

# Testing

TODO: add docker-compose setup for jaeger or similar.

Local process-compose testing hasn't been conclusive.

### CODE REVIEW

File: process-compose/suzuka-full-node/process-compose.telemetry.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: "3"
+
```
Comment (l-monninger) [inline process-compose/suzuka-full-node/process-compose.telemetry.yml:2]: If you can add an OTLP runner, that would be great.

Comment (l-monninger) [reply process-compose/suzuka-full-node/process-compose.telemetry.yml:2]: Our nix flake does package in Docker, so you can just spawn the container from process compose.

**Review by l-monninger — CHANGES_REQUESTED** [review]

File: process-compose/suzuka-full-node/process-compose.telemetry.yml (hunk 1 above)
Comment (l-monninger) [reply process-compose/suzuka-full-node/process-compose.telemetry.yml:2]:
```
just suzuka-full-node native build.setup.eth-local.celestia-local.test -t=false
```

File: protocol-units/execution/opt-executor/src/transaction_pipe.rs [hunk 2]
```diff
@@ -123,16 +131,14 @@ impl TransactionPipe {
...
 		let in_flight = self.transactions_in_flight.load(std::sync::atomic::Ordering::Relaxed);
-		info!(
-			target: "movement_timing",
-			in_flight = %in_flight,
-			"transactions_in_flight"
+		let otel_cx = OtelContext::current();
+		let otel_span = otel_cx.span();
+		otel_span.add_event(
+			"transactions_in_flight",
+			vec![KeyValue::new("in_flight", in_flight as i64)],
```
Comment (mzabaluev) [inline protocol-units/execution/opt-executor/src/transaction_pipe.rs:138]: This event looks like it wants to be a proper metric.

File: protocol-units/execution/opt-executor/src/executor/mod.rs [hunk 3]
```diff
@@ -46,7 +46,7 @@ impl Executor {
 		self.transactions_in_flight
 			.fetch_update(Ordering::Relaxed, Ordering::Relaxed, |current| {
 				info!(
-					target: "movement_timing",
+					target: "movement_telemetry",
 					count,
 					current,
 					"decrementing_transactions_in_flight",
```
Comment (mzabaluev) [inline protocol-units/execution/opt-executor/src/executor/mod.rs:52]: Also this one looks like a metric would be better (I haven't converted it to OpenTelemetry yet).

File: util/tracing/src/telemetry.rs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		Ok(Self { otlp_grpc_url })
+	}
+}
+
+/// Global initialization of the OpenTelemetry tracer provider.
+///
+/// This function should be called at the start of the program before any
+/// threads are able to use OpenTelemetry tracers. The function will panic
+/// if not called within a Tokio runtime.
+pub fn init_tracer_provider(
```
Comment (l-monninger) [inline util/tracing/src/telemetry.rs:50]: I would prefer these providers are fungible. Ideally, the content of a particularly (set) of environment variables, args, or other would tell an instance of the application which to use. How difficult is this to set up from where we are currently?

File: protocol-units/da/m1/light-node/src/v1/passthrough.rs [hunk 5]
```diff
@@ -146,7 +146,7 @@ impl LightNodeV1 {
 		Ok(verified_blobs)
 	}
 
-	#[tracing::instrument(target = "movement_timing", level = "debug")]
+	#[tracing::instrument(target = "movement_telemetry", level = "debug")]
```
Comment (l-monninger) [inline protocol-units/da/m1/light-node/src/v1/passthrough.rs:149]: I would like to have comments on each of these explaining why they are needed to achieve a certain metric as outlined here: 
- https://github.com/movementlabsxyz/movement/discussions/367
- https://github.com/movementlabsxyz/sre/discussions/63

**Review by l-monninger — CHANGES_REQUESTED** [review]

**Comment by l-monninger** [comment]
Will try to test this e2e shortly.

File: protocol-units/da/m1/light-node/src/v1/passthrough.rs (hunk 5 above)
Comment (mzabaluev) [reply protocol-units/da/m1/light-node/src/v1/passthrough.rs:149]: I'm not sure if we need all of these for telemetry. Originally they were added for perf research.
Do you think we can just remove those that don't contribute to a particular metric goal, or convert them to a plain spans/events with the default target?

File: util/tracing/src/telemetry.rs (hunk 4 above)
Comment (mzabaluev) [reply util/tracing/src/telemetry.rs:50]: As for fungibility in the API, tracing-opentelemetry was supposed to make OpenTelemetry exports just another sort of tracing macro call sites, but it does not work due to https://github.com/tokio-rs/tracing-opentelemetry/issues/159.
This is why I made `telemetry` a separate module with its own API.

**Comment by mzabaluev** [comment]
Superseded by #697.

**Pull request closed without merging**